packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -461,6 +461,9 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
   }
 
   bool canRequestFocus() {
+    if (widget.focusNode != null) {
+      return widget.focusNode!.canRequestFocus;
+    }
     if (widget.requestFocusOnTap != null) {
       return widget.requestFocusOnTap!;
     }
```

Comment:
[canRequestFocus](https://github.com/flutter/flutter/blob/ccb2b4439a430d4a736bcb294ba8f00be79fc66c/packages/flutter/lib/src/material/dropdown_menu.dart#L426-L435)  allows the text input field to be focused only on desktop unless `DropdownMenu.requestFocusOnTap` is set to `true`.  

This can be handled two ways:
1. A. focusable `FocusNode` overrides the default `requestFocusOnTap` value. (I added this)
2. Explicitly set `requestFocusOnTap` to `true` when providing a `FocusNode`.

cc: @QuncCccccc if you've any input here.

Reply:
Thanks for adding this! I think the first option makes more sense:)

Reply:
We should update the docs as well right? (To say which one wins.)

Reply:
Yep!

Reply:
pushed an update to the docs